src/index.ts

```diff
@@ -146,34 +116,42 @@ export default class Matic extends ContractsBase {
     return this.web3Client.send(txObject, _options)
   }
 
-  depositEther(
-    amount: BN | string,
-    options?: SendOptions,
-  ) {
-    if (options && (!options.from || !amount )) {
+  depositEther(amount: BN | string, options?: SendOptions) {
+    if (options && (!options.from || !amount)) {
       throw new Error('options.from or amount is missing')
     }
     return this.depositManager.depositEther(amount, options)
   }
 
-  approveERC20TokensForDeposit(
-    token: address,
-    amount: BN | string,
-    options?: SendOptions,
-  ) {
+  async depositDataByHash(txHash: string) {
+    this.childChainAddress = await this.registry.registry.methods.getChildChainAndStateSender().call()
```

Comment:
I don't think this would work. `getChildChainAndStateSender()` returns 2 addresses. We need to pick up the first one. Moreover, I'd suggest initializing the child chain address during matic.js initialisation - in `initialize()` in `index.ts`.

Did we test this?

Reply:
Yes, on next like I am using `this. childChainAddress[0]`